pkg/xds/experimental/client/cell.go

```diff
@@ -0,0 +1,82 @@
+// SPDX-License-Identifier: Apache-2.0
+// Copyright Authors of Cilium
+
+package client
+
+import (
+	"context"
+	"fmt"
+	"log/slog"
+
+	"github.com/cilium/cilium/pkg/node"
+
+	"github.com/cilium/hive/cell"
+	"github.com/cilium/hive/job"
+	corepb "github.com/cilium/proxy/go/envoy/config/core/v3"
+	"github.com/spf13/pflag"
+	"google.golang.org/grpc"
+	core_v1 "k8s.io/api/core/v1"
+)
+
+var Cell = cell.Module(
+	"xds-client",
+	"xDS client",
+
+	cell.Config(defaultConfig),
+	cell.Provide(NewXDSClient),
+)
+
+type Config struct {
+	ServerAddr string `mapstructure:"xds-server-address"`
+
+	Options
+}
+
+var defaultConfig = Config{
+	ServerAddr: "",
+	Options:    *Defaults,
+}
+
+func (conf Config) Flags(flags *pflag.FlagSet) {
+	flags.String("xds-server-address", conf.ServerAddr, "Address of xDS server")
+	flags.Bool("xds-use-sotw-protocol", conf.UseSOTW, "Use State Of The World, non-incremental version of xDS protocol")
+}
+
+type input struct {
+	cell.In
+
+	Config
+	JobGroup       job.Group
+	Log            *slog.Logger
+	LocalNodeStore *node.LocalNodeStore
+}
+
+func NewXDSClient(in input) (BaseLayer, error) {
+	if in.Config.ServerAddr == "" {
+		// Disabled
+		return nil, nil
+	}
+	localNode, err := in.LocalNodeStore.Get(context.TODO())
```

Comment:
thx, changes introduced. I'm working on e2e test

```
Also node labels may change at any time and I don't see this handling that properly. It would be better to observe LocalNodeStore rather than doing a one-time Get. OTOH, I guess one can argue that topology zone can't change at runtime... :)
```
yeah the idea was that topology zone will not change in the runtime. My understanding is that to change the zone cilium need to be physically moved to a different machine, then it needs to be restarted, right?

Reply:
Yep I would expect so.